cmd/optimize/events.go

```diff
@@ -460,6 +482,32 @@ func listRecommendations(flags *recommendationsCmdFlags) func(*cobra.Command, []
 			_, next_ok = data_set.Links["next"]
 		}
 
+		// extract recommendation optimizer ID + num to query and append blocker data from optimization_started events
+		for i := range recommendationRows {
+			blockerRows, err := getOptimizationBlockerData(recommendationRows[i], flags)
+			if err != nil {
+				return fmt.Errorf("Could not get optimization_started data association with recommendation: %w", err)
+			}
+
+			recommendationRows[i].Blockers = make([]string, 0)
+
+			// merge recommendation and blocker data
+			for _, row2 := range *blockerRows {
+				for attr, val := range row2.EventAttributes {
+					recommendationRows[i].EventAttributes[attr] = val
+
+					// extract the ID from the attribute string
+					splitAttr := strings.Split(attr, ".")
+					if len(splitAttr) > 3 {
+						blockerID := splitAttr[len(splitAttr)-2]
+						if !strings.Contains(strings.Join(recommendationRows[i].Blockers, ","), blockerID) {
+							recommendationRows[i].Blockers = append(recommendationRows[i].Blockers, blockerID)
+						}
+					}
+				}
+			}
+		}
```

Comment:
I think modifying the EventAttributes reduces the clarity and utility of the `-o yaml` output option. Instead, I would recommend creating a new struct `reccomendationRow` with the following form.

```
type reccomendationRow struct {
	eventRow
	StartedEventAttributes map[string]any
	BlockersPresent        string
        Blockers        []string
}
```

You can assign the `sliceToMap` result of the relevant optimization_started event to the StartedEventAttributes without further processing.

BlockersPresent is a string instead of bool to handle cases where the optimization_started event cannot be located/associated (in which case it is left empty)